util/expensivequery/expensivequery.go

```diff
@@ -73,12 +83,82 @@ func (eqh *Handle) Run() {
 				}
 			}
 			threshold = atomic.LoadUint64(&variable.ExpensiveQueryTimeThreshold)
+
+			instanceStats := &runtime.MemStats{}
+			runtime.ReadMemStats(instanceStats)
+			instanceMem := instanceStats.HeapAlloc
+			if err1 == nil && instanceMem > systemMem/10*8 {
+				// At least ten seconds between two recordings that memory usage is less than threshold (80% system memory).
+				// If the memory is still exceeded, only records once.
+				if time.Since(lastOOMtime) > 10*time.Second {
```

Comment:
Should we clean the old profiles and keep the files of a recent number for not spamming the file system?